tests/integration/aof-multi-part.tcl

```diff
@@ -0,0 +1,1065 @@
+                [r dbsize] > 0
+            } else {
+                fail "No write load detected."
+            }
+
+            # First AOFRW done
+            assert_aof_manifest_content $aof_manifest_filepath {
+                {file appendonly.aof.1.base.rdb seq 1 type b}
+                {file appendonly.aof.1.incr.aof seq 1 type i}
+            }
+
+            # Check we really have these files
+            assert_equal 1 [check_file_exist $aof_dirpath $aof_manifest_name]
+            assert_equal 1 [check_file_exist $aof_dirpath "${aof_basename}.1${::base_aof_sufix}${::rdb_format_suffix}"]
+            assert_equal 1 [check_file_exist $aof_dirpath "${aof_basename}.1${::incr_aof_sufix}${::aof_format_suffix}"]
+
+            r bgrewriteaof
+            waitForBgrewriteaof r
+
+            # The second AOFRW done
+            assert_aof_manifest_content $aof_manifest_filepath {
+                {file appendonly.aof.2.base.rdb seq 2 type b}
+                {file appendonly.aof.2.incr.aof seq 2 type i}
+            }
+
+            assert_equal 1 [check_file_exist $aof_dirpath $aof_manifest_name]
+            # Wait bio delete history 
+            wait_for_condition 1000 500 {
+                [check_file_exist $aof_dirpath "${aof_basename}.1${::base_aof_sufix}${::rdb_format_suffix}"] == 0 &&
+                [check_file_exist $aof_dirpath "${aof_basename}.1${::incr_aof_sufix}${::aof_format_suffix}"] == 0
+            } else {
+                fail "Failed to delete history AOF"
+            }
+            assert_equal 1 [check_file_exist $aof_dirpath "${aof_basename}.2${::base_aof_sufix}${::rdb_format_suffix}"]
+            assert_equal 1 [check_file_exist $aof_dirpath "${aof_basename}.2${::incr_aof_sufix}${::aof_format_suffix}"]
+
+            stop_write_load $load_handle0
+            wait_load_handlers_disconnected
+
+            set d1 [r debug digest]
+            r debug loadaof
+            set d2 [r debug digest]
+            assert {$d1 eq $d2}
+        }
+
+        test "AOF multiple rewrite failures will open multiple INCR AOFs" {
+            # Start write load
+            r config set rdb-key-save-delay 10000000
+
+            set orig_size [r dbsize]
+            set load_handle0 [start_write_load $master_host $master_port 10]
+
+            wait_for_condition 50 100 {
+                [r dbsize] > $orig_size
+            } else {
+                fail "No write load detected."
+            }
+
+            # Let AOFRW fail three times
+            r bgrewriteaof
+            set fork_child_pid [get_child_pid 0]
+            exec kill -9 $fork_child_pid
+            waitForBgrewriteaof r
+
+            r bgrewriteaof
+            set fork_child_pid [get_child_pid 0]
+            exec kill -9 $fork_child_pid
+            waitForBgrewriteaof r
+
+            r bgrewriteaof
+            set fork_child_pid [get_child_pid 0]
+            exec kill -9 $fork_child_pid
+            waitForBgrewriteaof r
```

Comment:
maybe use this opportunity to check that the temp files are deleted?
(by counting the number of files in the folder)

Reply:
Added